src/wh_12_3/SumOfNumbers.java

```diff
@@ -0,0 +1,83 @@
+package wh_12_3;
+
+import java.util.*;
+
+public class SumOfNumbers {
+    private static Scanner sc = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        List<Double> numbers = new ArrayList<>();
+//        fillListFromUser(numbers);
+        fillListFromUser(numbers);
+        try {
+            reversOrder(numbers);
+            sumOfNumbers(numbers);
+            maxAndMinNumbers(numbers);
+        } catch (IllegalArgumentException ex) {
+            System.out.println("Na pustej liscie nie da sie wykonac operacji");
+        }
+    }
+
+    static List<Double> fillListFromUser(List<Double> list) {
+        boolean input = true;
+        do {
+            System.out.println("Podaj kolejną liczbę dodatnia (podaj ujemna liczbe aby wyjsc z programu)");
+            double value = sc.nextDouble();
+            if (value > 0) {
+                list.add(value);
+            } else {
+                input = false;
+            }
+        }
+        while (input);
+        return list;
+    }
+
+//    static void fillListFromUser(List<Double> list) {
+//        System.out.println("Podaj kolejną liczbę dodatnia (podaj ujemna liczbe aby wyjsc z programu)");
+//        double input = sc.nextDouble();
+//        if (input < 0)
+//            return;
+//        try {
+//            list.add(input);
+//        } catch (NumberFormatException e) {
+//            System.err.println("Liczba w niepoprawnym formacie");
+//        }
+//        fillListFromUser(list);
+//    }
+
+    static void sumOfNumbers(List<Double> list) {
+        if (list.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+        StringBuilder sb = new StringBuilder();
+        int sum = 0;
+        for (double d : list) {
+            sb.append(d);
+            sb.append('+');
+            sum += d;
+        }
+        sb.replace(sb.length() - 1, sb.length(), "=");
+        sb.append(sum);
+        System.out.println(sb.toString());
+    }
+
+    static void reversOrder(List<Double> list) {
```

Comment:
mały tip dla metod jak `reversOrder` i `maxAndMinNumbers` lepiej zeby metody zwrocily wartosc ktora obliczaja. i zamaist wyswietlac wartosc w metodzie lepiej zrobic to poza metoda